Final Project/battleship.rb

```diff
@@ -0,0 +1,141 @@
+require 'colorize'
+
+def placeShip(originalBoard, length)
+	board = originalBoard
+	row = rand(10)
+	coloumn = rand(10)
+	if rand(2) == 0
+		if coloumn < 5
+			(0...length).each do |num|
+				board[row][coloumn + num] = "S".colorize(:light_cyan)
+			end
+		else
+			(0...length).each do |num|
+				board[row][coloumn - num] = "S".colorize(:light_cyan)
+			end
+		end
+	else
+		if row < 5
+			(0...length).each do |num|
+				board[row + num][coloumn] = "S".colorize(:light_cyan)
+			end
+		else
+			(0...length).each do |num|
+				board[row - num][coloumn] = "S".colorize(:light_cyan)
+			end
+		end
+	end
+	return board
+end
+
+def createBoard(ships)
+	newBoard = Array.new()
+	(1..10).each do |index|
+		newBoard.push([])
```

Comment:
Ahh, so you want to push 10 blank arrays. Okay! So for starters in Ruby, we don't actually use push.

```rb
newBoard << []
```

`<<` means push.

Secondly the way to refactor is as follows.

```rb
10.times.map { [] }
```

This means do 10 times. Map means replace index position with as it loops.

```ruby
ary = []
for i in 0..10
  ary[i] = []
end 
```